src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -31,7 +42,34 @@ public Task<OpenApiDocument> GetOpenApiDocumentAsync()
             Paths = GetOpenApiPaths(capturedTags),
             Tags = [.. capturedTags]
         };
-        return Task.FromResult(document);
+        await ApplyTransformers(document, cancellationToken);
+        return document;
+    }
+
+    private async Task ApplyTransformers(OpenApiDocument document, CancellationToken cancellationToken = default)
+    {
+        var documentTransformerContext = new OpenApiDocumentTransformerContext
+        {
+            DocumentName = documentName,
+            ApplicationServices = serviceProvider,
+            DescriptionGroups = apiDescriptionGroupCollectionProvider.ApiDescriptionGroups.Items,
+        };
+        for (var i = 0; i < _options.DocumentTransformers.Count; i++)
```

Comment:
Why not foreach?

Reply:
Avoiding allocating an enumerator in a hot codepath.

Reply:
Can you please add a comment to that effect so someone doesn't "fix" it?